modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java

```diff
@@ -172,6 +173,41 @@ Set<Property> getProperties() {
         return ipDataLookup.getProperties();
     }
 
+    /**
+     * Writes GeoIP data to the document. In flexible field access mode, writes individual dotted fields
+     * (e.g., "my.field.city", "my.field.country") instead of a single nested object. The "location" field
+     * is written as an array [lon, lat] for better compatibility with geo_point fields in flexible mode.
+     *
+     * @param document the ingest document
+     * @param targetField the base target field path
+     * @param data the GeoIP data to write
+     */
+    private void writeGeoIpData(IngestDocument document, String targetField, Map<String, Object> data) {
+        if (document.getCurrentAccessPatternSafe() == FLEXIBLE) {
+            // In flexible mode, write each property as a separate dotted field
+            for (Map.Entry<String, Object> entry : data.entrySet()) {
+                String key = entry.getKey();
+                Object value = entry.getValue();
+
+                // Convert location from map {lat, lon} to array [lon, lat] in flexible mode
+                if ("location".equals(key) && value instanceof Map) {
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> locationMap = (Map<String, Object>) value;
+                    Double lat = (Double) locationMap.get("lat");
+                    Double lon = (Double) locationMap.get("lon");
+                    if (lat != null && lon != null) {
+                        value = List.of(lon, lat); // GeoJSON order: [lon, lat]
+                    }
+                }
+
```

Comment:
I think it might make sense to `assert` here that the value is not an `instanceof Map`. My reasoning is that although all the current databases currently return flat maps except for the location, some future PR could add a new map-valued key and then this code would (as written) very quietly start doing the wrong thing.

It seems like it would be less trappy for us to alert that future developer at dev-time as soon as they go to run some tests of their new code.

Reply:
Good idea, added